### PR TITLE
chore: return show notification function

### PR DESCRIPTION
**Pull request by devceline**
# Description
**TL;DR: There's a race condition that causes the push notifications to be considered "silent push messages" which are flagged by the browser/OS as suspicious. Causing odd messages before every PN on desktop ("This site has been upgraded in the background") and lack of reliability on iOS. This should fix it.**

According to the sources below, not awaiting the `showNotification` and technically firing it after the listener resolves is a red flag for the browser/OS. 

- [StackOverflow, Chrome PNs causing odd messages "Site has been upgraded in the background"](https://stackoverflow.com/questions/31108699/chrome-push-notification-this-site-has-been-updated-in-the-background)
- [web.dev, handling push notifications](https://web.dev/articles/push-notifications-handling-messages#:~:text=Calling%20self.,the%20notification%20has%20been%20displayed.)
- [TruePush, iOS does not support silent web push messages](https://www.truepush.com/blog/ios-pwa-push-notifications/)
^ The means that a notification has to be shown to every `push` event. If the registration's `showMessage` is fired after that is resolved, it is considered a silent message. Red flag for iOS

**This also fixes issue causing white screens, by guarding every single reference to `Notification` even when it logically wouldn't be accessed.  Due to "Variable Not Found Errors".**

[Slack thread about Notification variable causing white screens](https://walletconnect.slack.com/archives/C0665P2HYM9/p1700230743240109?thread_ts=1700230313.710669&cid=C0665P2HYM9)
# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by bkrem — APPROVED** [review]

**Comment by bkrem** [comment]
Thanks for references/related reading here to help understand the problem 🙏
